CMakeLists.txt

```diff
@@ -1143,11 +1143,15 @@ configure_file(${CMAKE_SOURCE_DIR}/cmake/Makefile.tests ${CMAKE_BINARY_DIR}/incl
 configure_file(${CMAKE_SOURCE_DIR}/cmake/Makefile.tests.common ${CMAKE_BINARY_DIR}/include/Makefile.tests.common COPYONLY)
 
 # Configure CharmConfig.cmake
-configure_file(
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
   ${CMAKE_SOURCE_DIR}/cmake/templates/CharmConfig.cmake.in
   "${CMAKE_BINARY_DIR}/lib/cmake/Charm/CharmConfig.cmake"
-  ESCAPE_QUOTES
-  @ONLY)
+  INSTALL_DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/Charm/CharmConfig.cmake
+  )
+write_basic_package_version_file(
+  ${CMAKE_BINARY_DIR}/lib/cmake/Charm/CharmConfigVersion.cmake
+  COMPATIBILITY AnyNewerVersion)
```

Comment:
I'm not sure if all versions will be backward compatible. I'll let @matthiasdiener , @evan-charmworks and @rbuch to chime in on this one.

Reply:
I think we might want `SameMinorVersion` here.